### ISSUE TITLE
Track of ESLint Ignored Rules

### ISSUE THREAD

**Issue by fatadel** [eclipse/thingweb.node-wot#534]
This issue is created to keep track of ignored eslint rules with respective justifications.

1) **Flie/Line:** `https://github.com/eclipse/thingweb.node-wot/blob/master/packages/td-tools/src/thing-description.ts`
**Ignored Rule:** `no-use-before-define`
**Reason:** Many circular dependencies where interface/type/class A uses interface/type/class B in its definition and interface/type/class B uses interface/type/class A in its definition.

**Comment by JKRhb** [comment]
If I see it correctly, we are targeting ES6 with the TS transpiler at the moment, right? Then I think we could actually disable the rule `no-use-before-define` completely which apparently is only relevant if a JavaScript version prior to ES6 is used at the transpile target (see https://eslint.org/docs/rules/no-use-before-define).

**Comment by danielpeintner** [comment]
> This point has actually been resolved in the meantime, by the way :)

I have striked through it

**Comment by danielpeintner** [comment]
> If I see it correctly, we are targeting ES6 with the TS transpiler at the moment, right? Then I think we could actually disable the rule `no-use-before-define` completely which apparently is only relevant if a JavaScript version prior to ES6 is used at the transpile target (see https://eslint.org/docs/rules/no-use-before-define).

ES6 dates back to ECMAScript 2015.
Hence I think this is okay.

@relu91 any opinion? 

BTW, I noticed that in [.prettierrc.json](https://github.com/eclipse/thingweb.node-wot/blob/59fe3264bb4cfa19a3c8cddb0f1c68b4685a4a47/.prettierrc.json) we still use es5. Shall we change that too?
see https://github.com/eclipse/thingweb.node-wot/search?q=ES5

**Comment by relu91** [comment]
> ES6 dates back to ECMAScript 2015.
> Hence I think this is okay.

Good point, but I think is still a good practice to define stuff before using them (i.e. classes types etc. ). So I would be ok for leaving the rule as it is, but I would not stop you if you think differently :) 

> BTW, I noticed that in [.prettierrc.json](https://github.com/eclipse/thingweb.node-wot/blob/59fe3264bb4cfa19a3c8cddb0f1c68b4685a4a47/.prettierrc.json) we still use es5. Shall we change that too?
see https://github.com/eclipse/thingweb.node-wot/search?q=ES5

Yeah, I think we should.

**Comment by JKRhb** [comment]
> > ES6 dates back to ECMAScript 2015.
> > Hence I think this is okay.
> 
> Good point, but I think is still a good practice to define stuff before using them (i.e. classes types etc. ). So I would be ok for leaving the rule as it is, but I would not stop you if you think differently :)

Oh, yeah, you are right, I meant only typescript related definitions with the `no-use-before-define` rule :) I just noticed that there is a typescript-specific rule in `@typescript-eslint` that can be used instead of the default so we can actually get rid of the false positives (see [here](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-use-before-define.md)). I would open a PR for this shortly :)
